github/repos_contents.go

```diff
@@ -261,7 +261,7 @@ func (s *RepositoriesService) GetArchiveLink(ctx context.Context, owner, repo st
 		return nil, nil, err
 	}
 	resp.Body.Close()
-	if resp.StatusCode != http.StatusFound {
+	if !(resp.StatusCode == http.StatusFound || resp.StatusCode == http.StatusMovedPermanently) {
```

Comment:
How about: 

`if resp.StatusCode != http.StatusFound && resp.StatusCode != http.StatusMovedPermanently {`

here and in the test file?